app/controllers/cases/filters_controller.rb

```diff
@@ -33,10 +33,11 @@ def closed
       service = call_search_service(unpaginated_cases)
       @query = service.query
 
+      @maximum_records_for_download = 1000
       if service.error?
         flash.now[:alert] = service.error_message
       else
-        if download_csv_request?
+        if download_csv_request? and service.result_set.count < @maximum_records_for_download
```

Comment:
Tiny thing :) Probably should be `&&` instead of `and` here - unless you're intentionally using it for flow control e.g. https://www.synopsys.com/blogs/software-security/ruby-demystified-and-vs/